src/main/java/me/zeroeightsix/kami/gui/kami/KamiGUI.java

```diff
@@ -448,6 +450,56 @@ public void onTick() {
         entityLabel.setFontRenderer(fontRenderer);
         frames.add(frame);
 
+        /*
+         * Potion Effects
+         */
+        frame = new Frame(getTheme(), new Stretcherlayout(1), "Potions");
+        frame.setCloseable(false);
+        frame.setPinnable(true);
+        Label potionsLabel = new Label("");
+        potionsLabel.addTickListener(new TickListener() {
+            Minecraft mc = Minecraft.getMinecraft();
+
+            @Override
+            public void onTick() {
+                potionsLabel.setText("");
+
+                if(mc.player == null){
+                    return;
+                }
+
+                // loop through potions and add them to the potions label
+                mc.player.getActivePotionMap().forEach((potion, potionEffect) -> {
+                    String name = I18n.format(potion.getName());
+                    String amplifier;
+                    switch (potionEffect.getAmplifier()) {
+                        case 0:
+                            amplifier = "I";
+                            break;
+                        case 1:
+                            amplifier = "II";
+                            break;
+                        default:
+                            amplifier = "?";
+                            break;
+                    }
+                    int secDuration = potionEffect.getDuration() / 20; // 20 ticks per sec, this may be incorrect in laggy servers
+                    long min = TimeUnit.SECONDS.toMinutes(secDuration);
+                    long secs = TimeUnit.SECONDS.toSeconds(secDuration) - (min * 60);
+                    String potionText = String.format("%s %s (%d:%02d)", name, amplifier, min, secs);
+                    potionsLabel.addLine(potionText);
+                });
+
+            }
+        });
+
+        frame.addChild(potionsLabel);
+        potionsLabel.setFontRenderer(fontRenderer);
```

Comment:
if you create a frame and add lines you don't need to do this unless you want to GLColor it

Reply:
Alright! Should I remove it at all or just leave it for when you add potion coloring?

Reply:
I'll also do that myself, ty